spec/controllers/teachers/classroom_manager_controller_spec.rb

```diff
@@ -35,4 +39,422 @@
+      end
+
+      it 'should redirect to new teachers classroom path' do
+        get :lesson_planner, id: teacher.id
+        expect(response).to redirect_to new_teachers_classroom_path
+      end
+    end
+
+    context 'when classrooms i teach are not empty' do
+      it 'should assign the tab, grade, students, last_classroom_name and last_lassroom_id' do
+        get :lesson_planner, id: teacher.id, tab: "test tab", grade: "test grade"
+        expect(assigns(:tab)).to eq "test tab"
+        expect(assigns(:grade)).to eq "test grade"
+        expect(assigns(:students)).to eq user.students.any?
+        expect(assigns(:last_classroom_id)).to eq user.classrooms_i_teach.last.id
+        expect(assigns(:last_classroom_name)).to eq user.classrooms_i_teach.last.name
+      end
+    end
+  end
+
+  describe '#assign_activities' do
+    let!(:teacher) { create(:classrooms_teacher, user: user, role: "owner") }
+    let(:user) { create(:teacher, first_name: "test") }
+
+    before do
+      allow(controller).to receive(:current_user) { user }
+    end
+
+    context 'when current user is not staff and has no classrooms i teach' do
+      before do
+        allow(user).to receive(:classrooms_i_teach) { [] }
+      end
+
+      it 'should redirect to new teachers classroom path' do
+        get :assign_activities
+        expect(response).to redirect_to new_teachers_classroom_path
+      end
+    end
+
+    context 'when current user is staff or has classrooms i teach' do
+      context 'when user is staff' do
+        before do
+          user.role = "staff"
+        end
+
+        it 'should assign the tab, grade, students, last_classroom_name and last_classroom_id' do
+          get :assign_activities, tab: "test tab", grade: "test grade"
+          expect(assigns(:tab)).to eq "test tab"
+          expect(assigns(:grade)).to eq "test grade"
+          expect(assigns(:students)).to eq user.students.any?
+          expect(assigns(:last_classroom_id)).to eq user.classrooms_i_teach.last.id
+          expect(assigns(:last_classroom_name)).to eq user.classrooms_i_teach.last.name
+        end
+      end
+
+      context 'when user has classrooms i teach' do
+        it 'should assign the tab, grade, students, last_classroom_name and last_classroom_id' do
+          get :assign_activities, tab: "test tab", grade: "test grade"
+          expect(assigns(:tab)).to eq "test tab"
+          expect(assigns(:grade)).to eq "test grade"
+          expect(assigns(:students)).to eq user.students.any?
+          expect(assigns(:last_classroom_id)).to eq user.classrooms_i_teach.last.id
+          expect(assigns(:last_classroom_name)).to eq user.classrooms_i_teach.last.name
+        end
+      end
+    end
+  end
+
+  describe '#generic_add_students' do
+    let(:student) { create(:student) }
+    let(:teacher) { create(:teacher) }
+
+    # the before filter authorize_teacher! redirects the user so this branch is never traversed
```

Comment:
Can we write a test that confirms the controller action the routes is never called then?

Reply:
do you mean adding a test for the before filter?

Reply:
If possible, yes